model/model.proto

```diff
@@ -0,0 +1,206 @@
+    ONLINE             = 1;
+    ERROR              = 2;
+  }
+  string name        = 1 [(google.api.field_behavior) = REQUIRED];
+  string description = 2;
+  Status status      = 3;
+}
+
+message GetModelRequest {
+    string name    = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+message LoadModelRequest {
+    string name = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+message LoadModelResponse {
+    // OPTIONAL - If nontrivial cost is involved in
+    // determining the size, return 0 here and
+    // do the sizing in the modelSize function
+    uint64 sizeInBytes = 1;
+
+    // EXPERIMENTAL - Applies only if limitModelConcurrency = true
+    // was returned from runtimeStatus rpc.
+    // See RuntimeStatusResponse.limitModelConcurrency for more detail
+    uint32 maxConcurrency = 2;
+}
+
+message UnloadModelRequest {
+    string name = 1 [(google.api.field_behavior) = REQUIRED];
+}
+
+message UnloadModelResponse {}
+
+message PredictModelRequest {
+    enum CVTask {
+        CLASSIFICATION = 0;
+        DETECTION      = 1;
+    }
+    // model name
+    string name      = 1 [(google.api.field_behavior) = REQUIRED];
+    // model version
+    int32 version    = 2 [(google.api.field_behavior) = REQUIRED];
+    // byte array of image content
+    bytes content    = 3 [(google.api.field_behavior) = REQUIRED];
+    // CV task type such as object detection, classification
+    CVTask type      = 4 [(google.api.field_behavior) = REQUIRED];
+}
+
+message ListModelRequest {}
+
+message ListModelResponse {
+    repeated ModelInfo models = 1;
+}
+
+message CreateModelsResponse {
+    repeated ModelInfo models = 1;
+}
+
+message CreateModelRequest {
+  bytes content      = 1 [(google.api.field_behavior) = REQUIRED];
+  string type        = 2 [(google.api.field_behavior) = OPTIONAL];
+  string framework   = 3 [(google.api.field_behavior) = OPTIONAL];
+  string description = 4 [(google.api.field_behavior) = OPTIONAL];
+  bool optimized     = 5 [(google.api.field_behavior) = OPTIONAL];
+  string visibility  = 6 [(google.api.field_behavior) = OPTIONAL];
+}
+
+message ModelVersion {
+  int32 version        = 1;
+  int32 model_id       = 2;
+  string description   = 3;
+  google.protobuf.Timestamp created_at = 4;
+  google.protobuf.Timestamp updated_at = 5;
+  string status        = 6;
+}
+
+message ModelInfo {
+  int32 id                       = 1;
+  string name                    = 2;
+  bool optimized                 = 3;
+  string description             = 4;
+  string type                    = 5;
+  string framework               = 6;
+  google.protobuf.Timestamp created_at           = 7;
+  google.protobuf.Timestamp updated_at           = 8;
+  string organization            = 9;
+  string visibility              = 10;
+  repeated ModelVersion versions = 11;
+  string icon                    = 12;
+  string Author                  = 13;
+  string fullName                = 14 [(google.api.field_behavior) = OUTPUT_ONLY];
```

Comment:
```suggestion
  int32 id                       = 1;
  string name                    = 2;
  bool optimized                 = 3;
  string description             = 4;
  string type                    = 5;
  string framework               = 6;
  google.protobuf.Timestamp created_at           = 7;
  google.protobuf.Timestamp updated_at           = 8;
  string visibility              = 9;
  repeated ModelVersion versions = 10;
  string fullName                = 11 [(google.api.field_behavior) = OUTPUT_ONLY];
```

Reply:
@xiaofei-du yes I will add that later

Reply:
Please add in this PR to make it self-contained.

Reply:
@xiaofei-du cc @bochengyang  agreed, I already added that

Reply:
@Phelan164 Don't see any commit related to this

Reply:
@bochengyang commit in [this](https://github.com/instill-ai/protobufs/pull/4/commits/65fa5bec54ab8777cdd5df51fc50258da309fbb1)